src/app/components/audio-recordings/components/analyses-download/analysis-download-row.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, EventEmitter, Inject, Input, Output } from "@angular/core";
+import { AnalysisJobItemResultsService } from "@baw-api/analysis/analysis-job-item-result.service";
+import { rootPath } from "@components/audio-recordings/pages/analysis-results/analyses-results.component";
+import { AnalysisJob } from "@models/AnalysisJob";
+import { AnalysisJobItemResult } from "@models/AnalysisJobItemResult";
+import { AudioRecording } from "@models/AudioRecording";
+import { API_ROOT } from "@services/config/config.tokens";
+
+@Component({
+  selector: "baw-directory-row",
+  templateUrl: "analysis-download-row.component.html",
+  styleUrls: ["analysis-download-row.component.scss"],
+})
+export class AnalysesDownloadRowComponent {
+  public constructor(
+    @Inject(API_ROOT) public apiRoot: string,
+    public api: AnalysisJobItemResultsService,
+  ) { }
+
+  public open: boolean;
+  @Input() public item: AnalysisJobItemResult;
+  @Input() public analysisJob: AnalysisJob;
+  @Input() public audioRecording: AudioRecording;
+  @Input() public even: boolean;
+  @Output() public loadChildren = new EventEmitter<AnalysisJobItemResult>();
+
+  private subDirectoriesCount = (path: string) => path.split("/").length;
```

Comment:
In all honestly, I did this for code clarity. This can be easily changed to a regular function, and I won't use arrow functions in this way in the future.

Reply:
if you answer what "I want to use it as a lambda for functional programming" then that would have been fine.